### ISSUE TITLE
config_for assumes a hash value as environment config

### ISSUE THREAD

**Issue by henvo** [rails/rails#37800]
### Steps to reproduce
In my app I am keeping the allowed hosts inside a configuration yaml file:
```
# config/allowed_hosts.yml
development:
  - http://localhost:3001
  - http://localhost:3002

production:
  - https://my.example.app
  - https://my.admin.app
```
and in my `config/application.rb` I simply use:

```
config.allowed_hosts = config_for(:allowed_hosts)
```

### Expected behavior
Up until Rails 6 I was able to use arrays as a direct value of the environment keys.
For example:

```
# config/initializers/cors.rb
...
origins Rails.configuration.allowed_hosts
```

### Actual behavior
Since https://github.com/rails/rails/commit/b167d521ab873112ff1b98520f184d8088feb754 the value is assumed to be a hash. Which raises an error:

```
TypeError: no implicit conversion of Array into Hash
/usr/local/bundle/gems/railties-6.0.1/lib/rails/application.rb:232:in `merge'
/usr/local/bundle/gems/railties-6.0.1/lib/rails/application.rb:232:in `config_for'
```

Is this expected behavior? I did not find any documentation that states that the value must be a hash.

### System configuration
**Rails version**: 6.0.1

**Ruby version**: 2.5.3

**Comment by ChiangChu** [comment]
This Problem still occure in the main rails 6.0.3. So why closing it when this is still not fixed?

**Comment by jonathanhefner** [comment]
@ChiangChu The fix from #37850 has now been backported to `6-0-stable` branch in 7d8cc2feb99b60cf0d2c76415b8b923246cca4a0, and will be included in Rails 6.0.4.
